tests/test_es_query.py

```diff
@@ -0,0 +1,203 @@
+from nefertari.es_query import compile_es_query, _get_tokens, _build_tree, _attach_nested
+
+
+class TestESQueryCompilation(object):
+
+    def test_build_parse_tokens(self):
+        query_string = 'item:value OR item:value'
+        assert ['item:value', 'OR', 'item:value'] == _get_tokens(query_string)
+        query_string = '((item:value OR item:value) AND NOT (item:OR OR NOT item:NOT)) OR true:true'
+        assert _get_tokens(query_string) == ['(', '(', 'item:value', 'OR', 'item:value', ')',
+                                             'AND NOT', '(', 'item:OR', 'OR NOT', 'item:NOT',
+                                             ')', ')', 'OR', 'true:true']
+
+    def test_remove_top_level_brackets_if_needless(self):
+        query_string = '(item:value OR item:value)'
+        tokens = _get_tokens(query_string)
+        tree = _build_tree(tokens)
+        assert ['item:value', 'OR', 'item:value'] == tree
```

Comment:
Please follow our convention for equalities:
`actual == expected`
In this case, `tree` is actual, `['item:value', 'OR', 'item:value']` is expected